src/Configurator/AddLinesConfigurator.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+namespace Symfony\Flex\Configurator;
+
+use Composer\IO\IOInterface;
+use Symfony\Flex\Lock;
+use Symfony\Flex\Recipe;
+use Symfony\Flex\Update\RecipeUpdate;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+class AddLinesConfigurator extends AbstractConfigurator
+{
+    private const POSITION_TOP = 'top';
+    private const POSITION_BOTTOM = 'bottom';
+    private const POSITION_AFTER_TARGET = 'after_target';
+
+    private const VALID_POSITIONS = [
+        self::POSITION_TOP,
+        self::POSITION_BOTTOM,
+        self::POSITION_AFTER_TARGET,
+    ];
+
+    public function configure(Recipe $recipe, $config, Lock $lock, array $options = []): void
+    {
+        foreach ($config as $patch) {
+            if (!isset($patch['file'])) {
+                throw new \InvalidArgumentException(sprintf('The "file" key is required for the "add-lines" configurator for recipe "%s".', $recipe->getName()));
+            }
+
+            if (isset($patch['requires']) && !$this->isPackageInstalled($patch['requires'])) {
```

Comment:
should this be a string only or should we support a list of package names ?

Reply:
turning it into an array later would require using a different key, to avoid making recipes incompatible with older versions of Flex that would break badly when receiving something else than a string. So if we don't support an array already, we should at least properly handled unsupported values.